doc/dev/autograph.rst

```diff
@@ -526,6 +517,129 @@ array(22)
+>>> f(0.1)
+array(9.)
+
+Break and continue
+~~~~~~~~~~~~~~~~~~
+
+Within a while loop, control flow statements ``break`` and ``continue``
+are not currently supported. Usage will result in an error:
+
+
+>>> @qjit(autograph=True)
+... def f(x):
+...     while x < 5:
+...         if x < 0:
+...             continue
+...         x = x + x ** 2
+...     return x
+SyntaxError: 'continue' not properly in loop
+
+
+Updating and assigning variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As with for loops, while loops that update variables can also be converted with AutoGraph:
+
+>>> @qjit(autograph=True)
+... def f(x):
+...     while x < 5:
+...         x = x + 2
+...     return x
+>>> f(4)
+array(6.4)
+
+However, like with conditionals, a similar restriction applies: variables
+which are updated across iterations of the loop must have a JAX compileable
+type (Booleans, Python numeric types, and JAX arrays).
+
+You can also utilize temporary variables within a while loop:
+
+>>> @qjit(autograph=True)
+... def f(x):
+...     while x < 5:
+...         c = 2
+...         x = x + 2 * c
+...     return x
+>>> f(4)
+array(8.4)
+
+Temporary variables used inside a loop --- and that are **not** passed to a
+function within the loop --- do not have any type restrictions.
+
+Logical statements
+------------------
+
+AutoGraph has support for capturing logical statements that involve dynamic variables --- that is,
+statements involving ``and``, ``not``, and ``or`` that return booleans --- allowing them to be
+computed at runtime.
+
+>>> @qjit(autograph=True)
+... def f(x: float, y: float):
+...     a = x >= 0.0 and x <= 1.0
+...     b = not y >= 1.0
+...     return a or b
+>>> f(0.5, 1.1)
+array(True)
+>>> f(1.5, 1.6)
+array(False)
+
+Internally, logical statements are converted as follows:
+
+- ``x and y`` to ``jnp.logical_and(x, y)``
+- ``x or y`` to ``jnp.logical_or(x, y)``
+- ``not x`` to ``jnp.logical_not(x)``
+
+Note that there are a couple of important constraints and restrictions that must be
+considered when working with logical statements.
+
+All arguments must be dynamic
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Only cases where **all arguments to the logical statement are dynamic** (that is, dependent on
+runtime values) are captured and converted by AutoGraph. Cases where one or both of the arguments
+are static will result in the logical statement falling back to Python, and
+being interpreted at compile time.
+
+For example,
+
+>>> @qjit(autograph=True)
+... def f(x):
+...     return x and True
+TracerBoolConversionError: Attempted boolean conversion of traced array with shape float64[]..
+The error occurred while tracing the function f_1 at /tmp/__autograph_generated_file3sgpmu5h.py:6 for make_jaxpr. This concrete value was not available in Python because it depends on the value of the argument x.
+
+Here, ``x`` is dynamic, but the other argument is static. As a result, Python will attempt
+to evaluate this expression at compile time and fail.
```

Comment:
Should we include the reverse case and show that it succeeds (just for comparison, point out the different behaviour of builtin `and` which motivated the restriction to dynamic values only)?

```py
>>> @qjit(autograph=True)
... def f(x):
...     return True and x
```

Reply:
@dime10 I gave it a go, but couldn't find a satisfactory way of explaining this that doesn't delve too deeply into how Python `and` behaves 🤔 It felt a bit like it was distracting from the main reference info